src/logic/hooks/useGetRecommendedNonce.tsx

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { getRecommendedNonce } from '../safe/api/fetchSafeTxGasEstimation'
+import { getLastTxNonce } from '../safe/store/selectors/gatewayTransactions'
+
+type UseGetRecommendedNonce = (safeAddress: string) => number | undefined
+
+const useGetRecommendedNonce: UseGetRecommendedNonce = (safeAddress) => {
+  const lastTxNonce = useSelector(getLastTxNonce)
+  const storeNextNonce = lastTxNonce ? lastTxNonce + 1 : undefined
+
+  const [recommendedNonce, setRecommendedNonce] = useState<number | undefined>(storeNextNonce)
+
+  useEffect(() => {
+    const fetchRecommendedNonce = async () => {
+      try {
+        const recommendedNonce = await getRecommendedNonce(safeAddress)
+        setRecommendedNonce(recommendedNonce)
```

Comment:
Make sure to add an `isCurrent` check here to avoid any issues on unmount.

Reply:
Sorry, I don't understand the comment.

Reply:
[Here](https://github.com/gnosis/safe-react/blob/1fdaefce570906009504d99923dd3ad530218b38/src/routes/safe/components/Transactions/TxList/TxSingularDetails.tsx#L60)'s an example. It avoids race conditions.